relayer/src/link.rs

```diff
@@ -1426,34 +1429,42 @@ impl RelayPath {
         Ok(())
     }
 
-    /// Pulls out the operational elements with elapsed delay period and that can
+    /// Destination-chain fetching of operational data.
+    ///
+    /// Pulls out an operational elements with elapsed delay period and that can
     /// now be processed. Does not block: if no OD fulfilled the delay period (or none is
-    /// scheduled), returns immediately with `vec![]`.
-    fn try_fetch_scheduled_operational_data(
-        &mut self,
-    ) -> (Vec<OperationalData>, Vec<OperationalData>) {
-        // The first elements of the op. data vector contain the oldest entry.
-        // Remove and return the elements with elapsed delay.
-
-        let src_ods: Vec<OperationalData> = self
-            .src_operational_data
+    /// scheduled), returns immediately with `None`.
+    fn try_fetch_scheduled_operational_data_dst(&mut self) -> Option<OperationalData> {
```

Comment:
The design of these methods is reminiscent of iterators because they are essentially returning the next element until there are none, so it might be more pragmatic to implement them as custom iterators.